intel_extension_for_transformers/llm/runtime/graph/models/model_utils/model_types.h

```diff
@@ -151,6 +151,20 @@ struct model_layer {
   struct ne_tensor* v_cache;
 };
 
+typedef int32_t model_pos;
+typedef int32_t model_seq_id;
+
+struct kv_token_cell {
+  model_pos pos = -1;   // token idx (for rope)
+  model_pos delta = 0;  // token shift delta (pos += delta)
+};
+
+struct kv_seq_cell {
+  std::vector<kv_token_cell> token_cells;
+  model_seq_id seq_id = -1;
+  bool empty = true;
+};
```

Comment:
`kv_token_cell` and  `kv_seq_cell` may be useful in continuous batching. If not, I will remove it at the next PR about the scheduler.